tests/auto/aggregation/tst_aggregation.cpp

```diff
@@ -6659,6 +6659,46 @@ void tst_Aggregation::testSyncAdapter()
     foreach (const QContactId &id, allIds) {
         QVERIFY(originalIds.contains(id));
     }
+
+    // now add a new contact remotely, which has a name.
+    // remove the name locally, and modify it remotely - this will cause a "composed detail" modification update.
+    tsa.addRemoteContact(accountId, "John", "TsaFive", "555555", TestSyncAdapter::ImplicitlyModifiable);
+    tsa.performTwoWaySync(accountId);
+    QTRY_COMPARE(finishedSpy.count(), 11);
+    allIds = m_cm->contactIds(allSyncTargets);
+    QCOMPARE(allIds.size(), originalIds.size() + 2); // remote + aggregate
+    allContacts = m_cm->contacts(allSyncTargets);
+    QContact tsaFiveStc, tsaFiveAgg;
+    for (int i = allContacts.size() - 1; i >= 0; --i) {
+        const QContact &c(allContacts[i]);
+        if (originalIds.contains(c.id())) {
+            allContacts.removeAt(i);
+        } else {
+            bool isAggregate = c.relatedContacts(QStringLiteral("Aggregates"), QContactRelationship::Second).size() > 0;
+            if (isAggregate) {
+                tsaFiveAgg = c;
+            } else {
+                tsaFiveStc = c;
+            }
+        }
+    }
+    QVERIFY(tsaFiveAgg.id() != QContactId());
+    QVERIFY(tsaFiveStc.id() != QContactId());
+    // now remove the name on the local copy
+    QContactName tsaFiveName = tsaFiveStc.detail<QContactName>();
+    tsaFiveStc.removeDetail(&tsaFiveName);
+    QVERIFY(m_cm->saveContact(&tsaFiveStc));
+    // now modify the name on the remote server, and trigger sync.
+    // during this process, the local contact will NOT have a QContactName detail
+    // so the modification pair will be <null, newName>.  This used to trigger a bug.
+    tsa.changeRemoteContactName(accountId, "John", "TsaFive", "Jonathan", "TsaFive");
+    tsa.performTwoWaySync(accountId);
+    QTRY_COMPARE(finishedSpy.count(), 12);
```

Comment:
There's no verification of the final contact state here?

Also, is it easy to just follow on by performing the same pair of modifications to the reversed contacts, and verifying that that is also handled correctly?